synapse/eventbus.py

```diff
@@ -66,13 +56,21 @@ def unlink(self, func):
             bus.unlink( callback )
 
         '''
-        self._syn_weak_links.discard(func)
         if func in self._syn_links:
             self._syn_links.remove(func)
 
-    def on(self, name, func, weak=False):
+    def on(self, name, func, **filts):
         '''
-        Add a callback func to the SynCallBacker.
+        Add an event bus callback for a specific event with optional filtering.
+
+        Args:
+            name (str):         An event name
+            func (function):    A callback function to receive event tufo
+            filt (tuple):       A list of (prop,valu) tuples to filter events
+            **filts:            Optional filter values for the event tuple.
```

Comment:
should note that these are positive filters (ie. the key/value pairs must match on the event for the callback to occur)